clash-protocols-memmap/src/Protocols/MemoryMap/TypeDescription.hs

```diff
@@ -0,0 +1,150 @@
+-- SPDX-FileCopyrightText: 2024 Google LLC
+--
+-- SPDX-License-Identifier: Apache-2.0
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
```

Comment:
Add a comment why. (I'm guessing TemplateHaskell staging restrictions? Also mention why it's fine, e.g., "users are not supposed to import `Internal` modules".)